sql/src/main/java/io/crate/analyze/AnalyzedInsertStatement.java

```diff
@@ -103,6 +119,16 @@
             generatedColumns,
             defaultExpressionColumns,
             false);
+        this.returnValues = returnValues;
+        if (!outputNames.isEmpty() && !returnValues.isEmpty()) {
+            this.fields = new ArrayList<>(outputNames.size());
+            Iterator<Symbol> outputsIterator = returnValues.iterator();
+            for (ColumnIdent path : outputNames) {
+                fields.add(new Field(new TableRelation(tableInfo), path, outputsIterator.next()));
```

Comment:
```suggestion
                fields.add(new Field(this, path, outputsIterator.next()));
```

The `relation` of a `Field` must point to the relation who owns the field, which in this case here is the `AnalyzedInsertStatement`.

Reply:
`AnalyzedInsertStatement` is not a `AnalyzedRelation` so it can't be `this`. The `subQueryRelation` would of course work, but i assumed that the subquery relation is not the relation who owns the output field, so this is why I ended with this solution. What I am missing here ?

Reply:
Oh I see, I missed that.

Could you add a inline comment about that? We'll need to address that eventually, but given that we did the same for Update I guess it is okay to move along as is for now.

Reply:
Ok, will do. Thanks for the feedback. So, a `AnalyzedInsertStatement` should be a `Relation` in an ideal world ?

Reply:
Currently my thinking is that in an ideal world the `Field` wouldn't have this `relation` reference in its current form.

I'm not sure about `AnalyzedInsertStatement` being a `Relation` - or changing the hierarchy there in some other form. The focus would be more on changing `fields()` and `outputs` (or in this case, the returnValues`, so that only 1 is needed and not both.

But that's something for the future.